src/api/java/baritone/api/utils/ExampleBaritoneControl.java

```diff
@@ -688,18 +692,20 @@ private Goal parseGoal(String[] params) {
         Goal goal;
         try {
             BetterBlockPos playerFeet = ctx.playerFeet();
-            switch (params.length) {
+
+            // tests wether a dimension input exists because the switch has to be handled differently
+            switch (params.length < 1 || (isNumeric(params[params.length - 1]) || params[params.length - 1].startsWith("~")) ? params.length : params.length - 1) {
                 case 0:
                     goal = new GoalBlock(playerFeet);
                     break;
                 case 1:
-                    goal = new GoalYLevel(parseOrDefault(params[0], playerFeet.y));
+                    goal = new GoalYLevel(parseOrDefault(params[0], playerFeet.y, 1));
                     break;
                 case 2:
-                    goal = new GoalXZ(parseOrDefault(params[0], playerFeet.x), parseOrDefault(params[1], playerFeet.z));
+                    goal = new GoalXZ(parseOrDefault(params[0], playerFeet.x, Math.pow(8, ctx.world().provider.getDimensionType().getId() - getDimensionByName(params[params.length - 1]).getId())), parseOrDefault(params[1], playerFeet.z, Math.pow(8, ctx.world().provider.getDimensionType().getId() - getDimensionByName(params[params.length - 1]).getId())));
```

Comment:
move to a helper function

Reply:
So I add a function thar calculates how much to multiply to the helper function with DimensionType as parameters.